pkg/rabbit/setup.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbit
+
+import (
+	"github.com/NeowayLabs/wabbit"
+	wabbitamqp "github.com/NeowayLabs/wabbit/amqp"
+	"go.uber.org/zap"
+)
+
+const maxRetries = 5
+
+func SetupRabbitMQ(
+	connection wabbit.Conn,
+	channel wabbit.Channel,
+	brokerURL string,
+	retryNumber *int,
+	logger *zap.SugaredLogger) {
+	if *retryNumber >= maxRetries {
+		logger.Fatalf("Max retries (%d) reached, unable to setup RabbitMQ correctly", maxRetries)
```

Comment:
Once we reach maxRetries, we're failing the pod and then letting k8s handle re-running the pod with a backoff? I think that's fine, just confirming that's what we want.

Reply:
That's exactly what I'm aiming for, the next push should behave like that